File: eBibliotekaClient/src/app/data/interfaces/Account.ts
import { Image } from './Image';

export class Account {
  email: string;
  username: string;
  firstName: string;
  lastName: string;
  profilePicture?: Image;
  updatedAt: Date;
  createdAt: Date;

  /**
   *
   */
  constructor() {
    this.email = 'Učitavanje';
    this.username = 'Učitavanje';
    this.firstName = 'Učitavanje';
    this.lastName = 'Učitavanje';
    this.updatedAt = new Date();
    this.createdAt = new Date();
  }
}

export interface AccountUpdateVM {
  email: string;
  username: string;
  firstName: string;
  lastName: string;
}

export interface PasswordChangeVM {
  currentPassword: string;
  newPassword: string;
}

File: eBibliotekaServer/AuthModule/Models/Account.cs
using eBibliotekaServer.ImageModule.Models;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eBibliotekaServer.AuthModule.Models
{
    public class Account
    {
        [Key]
        public int ID { get; set; }
        public string Email { get; set; }
        [Required]
        public string Username { get; set; }
        [Required]
        public byte[] Password { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int? ProfileImageID { get; set; }
        [ForeignKey(nameof(ProfileImageID))]
        public Image ProfileImage { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}
